tests/unit/test_agent.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+# pylint: disable=protected-access
+"""Test for agent relations."""
+
+import secrets
+from unittest.mock import MagicMock, PropertyMock
+
+import ops.testing
+import pytest
+from charms.operator_libs_linux.v1 import systemd
+
+import service
+from charm import JenkinsAgentCharm
+
+agent_relation_data = {"url": "http://example.com", "jenkins-agent-0_secret": secrets.token_hex(4)}
```

Comment:
Thanks for the remarks about mutability. Initially I thought to avoid making this fixture and use a variable for simplicity. I'll move this to a fixture